### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by bitbaggi** [Azure/azure-sdk-for-java#36377]
I don't know if i do anything wrong.
After upgrading to 5.x for spring boot 3.0, nothing works anymore.

First the error of this [issue](https://github.com/Azure/azure-sdk-for-java/issues/36320).
Now after a few more bugs, i got this error:
`
[WARN] 17.08.2023 09:21:54  --- [ServletWebServerApplicationContext      ] :   Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.azure.spring.cloud.autoconfigure.implementation.context.AzureTokenCredentialAutoConfiguration': Unsatisfied dependency expressed through constructor parameter 0: No qualifying bean of type 'com.azure.spring.cloud.autoconfigure.implementation.context.properties.AzureGlobalProperties' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
[INFO] 17.08.2023 09:21:54  --- [StandardService                         ] :   Stopping service [Tomcat]
[ERROR] 17.08.2023 09:21:54  --- [LoggingFailureAnalysisReporter          ] :   

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 0 of constructor in com.azure.spring.cloud.autoconfigure.implementation.context.AzureTokenCredentialAutoConfiguration required a bean of type 'com.azure.spring.cloud.autoconfigure.implementation.context.properties.AzureGlobalProperties' that could not be found.

Action:

Consider defining a bean of type '
com.azure.spring.cloud.autoconfigure.implementation.context.properties.AzureGlobalProperties' in your configuration.
` 

I set the configuration for azure in the application properties, connection string and inputs/outputs...

Do i anything wrong?, Could it be that Spring hates me :D

**Comment by Netyyyy** [comment]
Hi @bitbaggi , thanks for reaching out.
Which Spring Cloud Azure version do you use, 5.3.0 or 5.4.0? Could you help provide a minimal project? thanks.

**Comment by bitbaggi** [comment]
Okay, found the reason, more likely a workaround.
I'm using version 5.4.0.

After manually adding a Bean for AzureGlobalProperties this error no longer occures.

```
@Bean
public AzureGlobalProperties azureGlobalProperties() {
    return new AzureGlobalProperties();
}
```

**Comment by Netyyyy** [comment]
> Okay, found the reason, more likely a workaround. I'm using version 5.4.0.
> 
> After manually adding a Bean for AzureGlobalProperties this error no longer occures.
> 
> ```
> @Bean
> public AzureGlobalProperties azureGlobalProperties() {
>     return new AzureGlobalProperties();
> }
> ```

Have you ever used Spring native or AOT?

**Comment by bitbaggi** [comment]
Yes, without the service bus dependency its working fine.

**Comment by Netyyyy** [comment]
> Yes, without the service bus dependency its working fine.

Do you mean that it's ok to use Spring Native and AOT, but only if add the `spring-cloud-azure-starter-servicebus`, it failed to start?

**Comment by bitbaggi** [comment]
Yeah, when i add the service bus dependency, spring stops to start without this bean
```
@Bean
public AzureGlobalProperties azureGlobalProperties() {
    return new AzureGlobalProperties();
}
```

**Comment by apahne** [comment]
Pretty much seeing the same error message when trying to build a native image that uses `spring-cloud-azure-starter-active-directory`. Using 5.4.0.

The workaround does it for me, too.
